File: src/test/java/leetcode75/lvl1/RunningSumOf1dArrayTest.java
package leetcode75.lvl1;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class RunningSumOf1dArrayTest {

    @Test
    void runningSum() {
        RunningSumOf1dArray task = new RunningSumOf1dArray();

        int[] expected = new int[] {0,0,0,0,0,0};
        int[] actual = new int[] {0,0,0,0,0,0};
        Assertions.assertArrayEquals(expected, task.runningSum(actual));

        expected = new int[] {1, 5, 10, 13};
        actual = new int[] {1, 4, 5, 3};
        Assertions.assertArrayEquals(expected, task.runningSum(actual));

        expected = new int[] {1, 3, 6, 10};
        actual = new int[] {1, 2, 3, 4};
        Assertions.assertArrayEquals(expected, task.runningSum(actual));

        expected = new int[] {1, 3, 6, 10};
        actual = new int[] {1, 2, 3, 4};
        Assertions.assertArrayEquals(expected, task.runningSum(actual));
    }
}

File: src/test/kotlin/task/kotlin/leetcode75/lvl1/RunningSumOf1dArrayTest.kt
package task.kotlin.leetcode75.lvl1

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class RunningSumOf1dArrayTest {
    private val testClass: RunningSumOf1dArray = RunningSumOf1dArray()

    @Test
    fun runningSumTest() {
        // case: 1
        var arr = intArrayOf(1,2,3,4)
        var expected = intArrayOf(1,3,6,10)

        Assertions.assertThat(testClass.runningSum(arr)).isEqualTo(expected)

        // case: 2
        arr = intArrayOf(1,1,1,1,1)
        expected = intArrayOf(1,2,3,4,5)

        Assertions.assertThat(testClass.runningSum(arr)).isEqualTo(expected)

        // case: 3
        arr = intArrayOf(3,1,2,10,1)
        expected = intArrayOf(3,4,6,16,17)

        Assertions.assertThat(testClass.runningSum(arr)).isEqualTo(expected)
    }
}